nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -234,4 +361,10 @@ public void process(final OutputStream out) throws IOException {
             }
         }
     }
+
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeChange(final PrimaryNodeState newState) {
+        isPrimaryNode = (newState == PrimaryNodeState.ELECTED_PRIMARY_NODE);
```

Comment:
I am running a secure cluster and when I have "primary node only" configured for the sending of the messages they never get sent. I believe this is because the primary node has been the same since the processor was added and it was never changed, therefore this method would never have gotten called. 

I think this needs some way to know that the processor is running on the primary node when it gets scheduled or initialized. I'm not sure if that is possible at the moment though.